lib/net/ssh/authentication/session.rb

```diff
@@ -43,6 +46,7 @@ def initialize(transport, options={})
       @transport = transport
 
       @auth_methods = options[:auth_methods] || Net::SSH::Config.default_auth_methods
+      @default_user = options[:default_user] || Etc.getlogin
```

Comment:
I was unsure whether the preference is reading `Etc.getlogin` directly, or somehow via a similar `Net::SSH::Config` setting?